### ISSUE TITLE
Long execution time (ImageMagick 7.1.0-21 and later)

### ISSUE THREAD

**Issue by usarise** [Imagick/imagick#538]
### Test script:
```php
<?php

\define('TEST_START', microtime(true));

$draw = new \ImagickDraw();
$draw->setFillColor('#000000');

for ($x = 0; $x < 10000; $x++) {
    $draw->point(rand(0, 500), rand(0, 500));
}

$imagick = new \Imagick();
$imagick->newImage(500, 500, new \ImagickPixel('#ffffff'));
$imagick->setImageFormat('png');
$imagick->drawImage($draw);

echo number_format((microtime(true) - TEST_START), 2) . "\n";
```
### Example execution time:

ImageMagick 7.1.0-20 and below.
**0.13 second.**

ImageMagick 7.1.0-21 and later.
**35.93~ second.**

The problem arose presumably after this commit https://github.com/ImageMagick/ImageMagick/commit/1496901aea4d6de5bc5c1051a423e95a80d77995

**Comment by Danack** [comment]
Thanks. Reported upstream.

Out of interest, how did you find the commit that was the issue? git bisect? If you have a script that automated it, I'd love to have a copy of it for reference.

**Comment by usarise** [comment]
Via `git diff 7.1.0-20 7.1.0-21 --stat` saw the filename `MagickCore/draw.c`

I opened the file history, saw that this year there is only one commit.
https://github.com/ImageMagick/ImageMagick/commits/7.1.0-21/MagickCore/draw.c
I tried to rollback the changes for version 7.1.0-22 and the problem is gone.

**Comment by usarise** [comment]
Fixed in ImageMagick-7.1.0-23
